src/SamplesApp/UITests.Shared/Windows_Devices/Midi/MidiDeviceOutput.xaml.cs

```diff
@@ -0,0 +1,635 @@
+			switch (this.currentMessageType)
+			{
+				case MidiMessageType.ProgramChange:
+				case MidiMessageType.ChannelPressure:
+				case MidiMessageType.PitchBendChange:
+				case MidiMessageType.MidiTimeCode:
+
+					if (parameter2SelectedIndex != -1)
+					{
+						this.sendButton.IsEnabled = true;
+					}
+					break;
+
+				default:
+					this.sendButton.IsEnabled = false;
+					break;
+			}
+
+			switch (this.currentMessageType)
+			{
+				case MidiMessageType.NoteOff:
+				case MidiMessageType.NoteOn:
+				case MidiMessageType.PolyphonicKeyPressure:
+					// This list is for Notes, of which there are 128
+					PopulateParameterList(this.parameter2, 128, "Note");
+					break;
+
+				case MidiMessageType.ControlChange:
+					// This list is for Controllers, of which there are 128
+					PopulateParameterList(this.parameter2, 128, "Controller");
+					break;
+
+				case MidiMessageType.ProgramChange:
+					// This list is for Program Numbers, of which there are 128
+					PopulateParameterList(this.parameter2, 128, "Program Number");
+					break;
+
+				case MidiMessageType.ChannelPressure:
+					// This list is for Pressure Values, of which there are 128
+					PopulateParameterList(this.parameter2, 128, "Pressure Value");
+					break;
+
+				case MidiMessageType.PitchBendChange:
+					// This list is for Pitch Bend Values, of which there are 16384
+					PopulateParameterList(this.parameter2, 16384, "Pitch Bend Value");
+					break;
+
+				case MidiMessageType.MidiTimeCode:
+					// This list is for Values, of which there are 16
+					PopulateParameterList(this.parameter2, 16, "Value");
+					break;
+
+				default:
+					// Start with a clean slate
+					// Hide the first parameter
+					this.parameter2.Header = "";
+					this.parameter2.IsEnabled = false;
+					this.parameter2.Visibility = Visibility.Collapsed;
+					break;
+			}
+		}
+
+		/// <summary>
+		/// For MIDI message types that need the third parameter, populate the list
+		/// based on the message type. For message types that don't need the third
+		/// parameter, empty and hide it
+		/// </summary>
+		private void UpdateParameterList3()
+		{
+			// Find the index of the user's choice
+			int parameter3SelectedIndex = (int)this.parameter3.Value;
+
+			// The last set of MIDI message types don't need additional parameters
+			// For them, show the Send button as soon as user selects parameter 3 value from the list
+			// Set default to disable Send button for any message types that fall through
+			switch (this.currentMessageType)
+			{
+				case MidiMessageType.NoteOff:
+				case MidiMessageType.NoteOn:
+				case MidiMessageType.PolyphonicKeyPressure:
+				case MidiMessageType.ControlChange:
+
+					if (parameter3SelectedIndex != -1)
+					{
+						this.sendButton.IsEnabled = true;
+					}
+					break;
+
+				default:
+					this.sendButton.IsEnabled = false;
+					break;
+			}
+
+			switch (this.currentMessageType)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This condition was just checked on line 568.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5305560)